src/flags/sorting.rs

```diff
@@ -426,31 +397,34 @@ mod test_sort_order {
         assert_eq!(None, SortOrder::from_config(&Config::with_none()));
     }
 
-    #[test]
-    fn test_from_config_empty() {
-        let yaml_string = "---";
-        let yaml = YamlLoader::load_from_str(yaml_string).unwrap()[0].clone();
-        assert_eq!(None, SortOrder::from_config(&Config::with_yaml(yaml)));
-    }
-
     #[test]
     fn test_from_config_default() {
-        let yaml_string = "sorting:\n  reverse: false";
-        let yaml = YamlLoader::load_from_str(yaml_string).unwrap()[0].clone();
         assert_eq!(
             Some(SortOrder::Default),
-            SortOrder::from_config(&Config::with_yaml(yaml))
+            SortOrder::from_config(&Config::default())
```

Comment:
This test will break, if the default value in the `Config` changes, even though the implementation is still correct. The config variant should rather be specified directly in the test setup.

Reply:
Sorry, I did not get the point. could you please explain more about the change?

we use this test to make sure we are returning the right `default` value from config, and could be use in sorting, right?
IMO, It work as expected that failed if the default value in config changed.

Reply:
Your confusion here likely comes from the fact, that the SortOrder enum has a variant called "Default", however is has nothing to do with any real default, it is just a name. So if someone suddenly decides that the default value in the Config is now the "Reverse" variant instead of the "Default" variant, your test will immediately start failing, even though the method you are testing (SortOrder::from_config) is still working as expected. You should rather set up an empty Config in the test setup and and set the SortOrder to the "Default" variant explicitly.

Reply:
I was thinking that this test case should make sure the default config should match the default sorting value,
so if someone changed the `Default`, then he should also change the config file part.
the case should fail, if config file part missed.

maybe the test case name is not that obvious, I changed it to `test_from_config_default_config`

Reply:
I mean sure, but in that case please also add a test case where the reverse is set to be explicitly `Some(false)`, just as there already is a test with `Some(true)`.